CatAnalyzer/plugins/ttbbLepJetsAnalyzer.cc

```diff
@@ -732,10 +788,44 @@ void ttbbLepJetsAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetu
     // DR 
     b_DRAddJets = genttbarConeCat->begin()->dRaddJets();
 
+    // adding b jet four-momentum having minimum deltaR
+    std::vector<math::XYZTLorentzVector> genbjets = genttbarConeCat->begin()->bJets();
+    tmp2->Fill(genbjets.size());
+    double mindR=999;
+    int index1=0, index2=0;
+    for(unsigned int i=0; i < genbjets.size(); ++i){
+      if( genbjets[i].Pt() < 20 || std::abs(genbjets[i].Eta()) > 2.5 ) continue;
+      for(unsigned int j=0; j < genbjets.size(); ++j){
+	if( genbjets[j].Pt() < 20 || std::abs(genbjets[j].Eta()) > 2.5 ) continue;
+	if( i == j ) continue;
+	double tmp = reco::deltaR(genbjets[i], genbjets[j]);
+	if( tmp < mindR ){
+	  mindR = tmp;
+	  index1 = i;
+	  index2 = j;
+	}
+      }
+    }
+
+    b_mindRbjet1_pt = genbjets[index1].Pt();
+    b_mindRbjet1_eta = genbjets[index1].Eta();
+    b_mindRbjet1_phi = genbjets[index1].Phi();
+    b_mindRbjet1_e = genbjets[index1].E();
+
+    b_mindRbjet2_pt = genbjets[index2].Pt();
+    b_mindRbjet2_eta = genbjets[index2].Eta();
+    b_mindRbjet2_phi = genbjets[index2].Phi();
+    b_mindRbjet2_e = genbjets[index2].E();
+
+    b_mindR = mindR;
+
     if(genttbarConeCat->begin()-> NaddbJets20() > 1){
       EventInfo->Fill(2.5, 1.0); // Number of ttbb Events	
       if(genttbarConeCat->begin()->semiLeptonic(-1)) EventInfo->Fill(3.5, 1.0); // Number of ttbb Events (Includes tau) 
-      if(genttbarConeCat->begin()->semiLeptonic(0)) EventInfo->Fill(4.5, 1.0); // Number of ttbb Events (Includes tau leptonic decay)
+      if(genttbarConeCat->begin()->semiLeptonic(0)){
+	  EventInfo->Fill(4.5, 1.0); // Number of ttbb Events (Includes tau leptonic decay)
+          tmp1->Fill(0.5,1.0);  
```

Comment:
Filling with weight=1 ... is it intended? ignore genWeight?

Reply:
BTW, tmp1 is a bad idea for a variable name. give a meaningful name unless it is really a temporary variable used locally. I think this tmp1 starts from the very first part of this module.

Reply:
tmp1 was made to check the events which are categorized as ttbj but are not categorized as ttbj. We already found the reason and solved, tmp1 is not anymore used. I will remove this tmp1.